woocommerce/woocommerce-gutenberg-products-block
# Fix product reviews schema date fields to use new (WP 5.5) `date-time` format

Fixes #3005

This PR fixes the schema for product reviews API endpoint. 

In WordPress 5.5, the `type` [must be one of the primitive types](https://developer.wordpress.org/rest-api/extending-the-rest-api/schema/#primitive-types); previously we had `date-time`. 

Schemas can [declare the intended format of `type: string`](https://developer.wordpress.org/rest-api/extending-the-rest-api/schema/#format), and for our dates we should use `date-time` aka [RFC3339](https://tools.ietf.org/html/rfc3339#section-5.8).

So, this PR uses `type: string, format: date-time` for the two review date fields. 

I did a quick search to see if there were other API dates that should use this format - looks like this is the only API affected.

To test this I added some reviews and an `All Reviews` block and checked the response. 

### Screenshots

<img width="885" alt="Screen Shot 2020-09-03 at 3 57 51 PM" src="https://user-images.githubusercontent.com/4167300/92069790-64646a80-edfe-11ea-9bd3-e6738cc9b0e5.png">

### How to test the changes in this Pull Request:
1. Add some reviews.
2. Add an all reviews block.
3. View on front end and inspect API response, compare with schema.

I only tested with WordPress 5.5 - should we test with 5.3 or 5.4 as well? 

Also keen to confirm this fixes the `doing it wrong` error - is there a way to reproduce that (on `main`/without these changes)? Or is this only validated when POSTing?

### Changelog

> dev: Corrected schema for date fields in product reviews API.
